### PR TITLE
Post Interaction, Searching, Feed, and Profile

### PR DESCRIPTION
**Pull request by racecn**
Added:
- ability to filter feed via age, popularity
- ability to mute and unmute users via feed, and profile pages
- visual feedback to users upon performing a task
- basis for future notification system
- improved comments and logging
- improved followers/following list

Removed:
- some unnecessary logging
- typoes

Fixed:
- profile picture deletion
- username changes now delete old username document BEFORE changing the username, previously they were changing the username, and then going to delete the current username document, which was causing a 'storage leak' or documents that wouldn't ever be touched again.
- halved the amount of DB calls on some functions

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -31,6 +34,7 @@ const SearchResults = React.lazy(() => import('./pages/SearchResults'));
 const App = () => {
   return (
     <Router>
+      <ToastContainer />
```
Comment (rishithab48) [inline src/App.js:37]: Will this new ToastContainer component post the notification on the top of the page just like you showed in our meeting?

File: src/components/Posts/ReportOptions.js
Comment (rishithab48) [inline src/components/Posts/ReportOptions.js:1]: Nice handling of errors.

**Review by rishithab48 — APPROVED** [review]
Checked the branch. Everything runs well.

File: src/App.js (hunk 1 above)
Comment (racecn) [reply src/App.js:37]: Yes! exactly like in our meeting

**Pull request merged**